### PR TITLE
Fixes #33463 - Add RHUI ACS create wizard

### PR DESCRIPTION
**Pull request by lfu**
#### What are the changes introduced in this pull request?
Create a new ACS wizard for RHUI ACSs that is the same as Custom, but includes help text and copy/paste buttons specifically for RHUI.  RHUI ACS instructions are in this [KCS](https://access.redhat.com/articles/6972161#highlighted-use-case-1-sync-a-capsule-from-an-alternate-source-3).

Also, add descriptions for the different ACS types on their cards.

### CODE REVIEW

File: webpack/scenes/AlternateContentSources/Create/Steps/AcsUrlPaths.js
Comment (ianballou) [inline webpack/scenes/AlternateContentSources/Create/Steps/AcsUrlPaths.js:95]: I'd instead put this on the "Credentials" part of the wizard under the "Content credentials" bullet.

I think we could make the "Content credentials" bullet the default selection too. I don't know when someone would connect to rhui with basic auth.  In fact -- I'm pretty sure we can remove the basic auth section ("Manual authentication") entirely.

image

File: webpack/scenes/AlternateContentSources/Create/Steps/AcsUrlPaths.js
Comment (ianballou) [inline webpack/scenes/AlternateContentSources/Create/Steps/AcsUrlPaths.js:99]: I think "in the server" would be implied by Content -> Content Credentials
```suggestion
            {__('Create content credentials with the generated SSL certificate and key via Content -> Content Credentials.')}
```

File: webpack/scenes/AlternateContentSources/Create/Steps/AcsUrlPaths.js
Comment (ianballou) [inline webpack/scenes/AlternateContentSources/Create/Steps/AcsUrlPaths.js:42]:
```suggestion
        {__('Find the relative path for each RHUI repository and combine them in a comma-separated list.')}
```

File: webpack/scenes/AlternateContentSources/Create/Steps/AcsUrlPaths.js
Comment (ianballou) [inline webpack/scenes/AlternateContentSources/Create/Steps/AcsUrlPaths.js:38]:
```suggestion
        {__('On the RHUA Instance, check the available repositories.')}
```

File: webpack/scenes/AlternateContentSources/Create/Steps/AcsUrlPaths.js
Comment (ianballou) [inline webpack/scenes/AlternateContentSources/Create/Steps/AcsUrlPaths.js:95]: We've been standardizing around adding periods, so I've added some suggestions.

```suggestion
            {__('Generate certificates for the chosen repositories.')}
```

**Review by ianballou — COMMENTED** [review]
Some first observations:

1) I realize that the ACS type can still be chosen with RHUI.  Since File won't ever be used, we can force the user to use the Yum type
2) In the Base URL, file:// is allowed. We can take that out since rhui will always be downloaded over http/https

File: webpack/scenes/AlternateContentSources/Create/Steps/SelectSource.js
Comment (ianballou) [inline webpack/scenes/AlternateContentSources/Create/Steps/SelectSource.js:28]: I think we could actually hide the content type selection entirely since Yum will be the only option.

File: webpack/scenes/AlternateContentSources/Create/Steps/ACSCredentials.js
Comment (ianballou) [inline webpack/scenes/AlternateContentSources/Create/Steps/ACSCredentials.js:42]: `verifySSL` doesn't need to be true here. That's for peer validation rather than authentication with the server.  (In fact, the original Pulp KCS says to keep it false -- I'll need to double check that it's even okay for it to be true)

**Review by ianballou — COMMENTED** [review]
Some of the header help text will need to change:

1) URL and paths:
Enter in the base path and any subpaths that should be searched for alternate content. The base path can be a web address or a filesystem location.

to

Enter in the base path and any subpaths that should be searched for alternate content. The base path must be a web address pointing to the root RHUI content directory.

and

Enter basic authentication information or choose content credentials if required for this source.

to

Choose content credentials if required for this RHUI source.

**Comment by ianballou** [comment]
Also, for some reason, in the review details page, the source type is showing up as "Rhui". It looks a little funny being capitalized like a normal word:

image

File: webpack/scenes/AlternateContentSources/Create/Steps/ACSCredentials.js
Comment (lfu) [reply webpack/scenes/AlternateContentSources/Create/Steps/ACSCredentials.js:42]: > Also, for some reason, in the review details page, the source type is showing up as "Rhui"...

This was just implemented in https://github.com/Katello/katello/pull/10289.

**Comment by lfu** [comment]
> Also, for some reason, in the review details page, the source type is showing up as "Rhui". It looks a little funny being capitalized like a normal word:

This was just implemented in https://github.com/Katello/katello/pull/10289.

**Comment by ianballou** [comment]
This is looking great! Now we just need an ACK from UX.

**Comment by ianballou** [comment]
Notes from UX meeting:

Content type: turn into a "disabled dropdown" (greyed out)

URL and paths
  - Step display on left is wrong for some reason

  - Red warning under Base URL shouldn't show up before the users types anything in
    - Need another description about the URL? 
      - As a popover or a subtext perhaps (Maria needs to check on this)
    - Error should happen only when they are done typing
      - After a pause and/or moving the cursor away
  - Put base URL above the new help text to improve the flow (since users will enter the Base URL and then Subpaths)

Content credentials
  - Put credentials information in the very first step with something that draws their attention
    - ...because once a user sees that they need to create content credentials, they've already wasted time on entering the previous information. The page will need to be reloaded to see the newly created content credentials.

Maria will help us get new suggestions for the ACS Type descriptions and a draft for the first page with the content credentials information (since we want it to really pop and look nice).

File: webpack/scenes/AlternateContentSources/Create/Steps/SelectSource.js
Comment (ianballou) [inline webpack/scenes/AlternateContentSources/Create/Steps/SelectSource.js:122]:
```suggestion
          <Alert variant="info" title={__('Generate RHUI certificates for the desired repositories as necessary.')} />
```

**Review by ianballou — COMMENTED** [review]
Looking good! Just a wording description for now

File: webpack/scenes/AlternateContentSources/Create/Steps/SelectSource.js
Comment (ianballou) [inline webpack/scenes/AlternateContentSources/Create/Steps/SelectSource.js:127]: From our meeting, the main improvement suggested was adding a link here. What do you think about having the following text be an entire link?

"Create content credentials with the generated SSL certificated and key."

**Review by ianballou — COMMENTED** [review]
We'll get some final word on what Verify SSL does and then see if any improvements can be done here to make it more clear.

**Comment by ianballou** [comment]
So Verify SSL essentially checks that the server is who we think they are. The CA cert is required for that to work.  Samir's recent PR moves Verify SSL and the CA Cert together, which I think is enough.

File: webpack/scenes/AlternateContentSources/Create/Steps/SelectSource.js
Comment (lfu) [reply webpack/scenes/AlternateContentSources/Create/Steps/SelectSource.js:127]: Good idea.

**Comment by ianballou** [comment]
I think this is looking great! We just need one more pass by UX and then we should be good to go.

**Comment by ianballou** [comment]
I want to take another shot at the ACS type descriptions:

Custom: "Define any web or filesystem path as an alternate source for syncing content."

Simplified: "Sync smart proxy content directly from upstream repositories by simply selecting the desired products."

RHUI: "Guided steps for syncing smart proxy content from RHUI."

**Comment by ianballou** [comment]
[test katello]

**Comment by jeremylenz** [comment]
> I want to take another shot at the ACS type descriptions:
> 
> Custom: "Define any web or filesystem path as an alternate source for syncing content."
> 
> Simplified: "Sync smart proxy content directly from upstream repositories by simply selecting the desired products."
> 
> RHUI: "Guided steps for syncing smart proxy content from RHUI."

this is great! Maybe it would help if we break it up further: (a) what's an ACS; and (b) where does it come from. Like have a blurb at the top

Alternate Content Sources allow you to define an alternate source for syncing content in the event the primary server is not available.

Custom: "Define any web or filesystem path."

Simplified: "Sync smart proxy content directly from upstream repositories by simply selecting the desired products."

RHUI: "Guided steps for syncing smart proxy content from RHUI."

**Comment by ianballou** [comment]
@jeremylenz I like the idea of having a blurb up top.  "...in the event the primary server is not available" interestingly isn't the main use case -- the alternate content source actually becomes the main content source.  It's more for importing content that is closer to you.  Like an override.  It's too late now, but I wonder if it should've been called override content source...

Anyway, here's my idea for the top blurb:  "Alternate content sources define new locations to download content from at repository or smart proxy sync time."

Then, Custom could become: "Define web or filesystem paths hold repositories."
  -> I'm making this more specific because refresh will fail if the path doesn't actually contain a repository that Pulp understands.

**Comment by ianballou** [comment]
Also, I want to change RHUI because it's the only one that doesn't start with a verb:

"Sync smart proxy content from RHUI with guided steps"

**Comment by ianballou** [comment]
So, altogether:

New top blurb:  "Alternate content sources define new locations to download content from at repository or smart proxy sync time."

Custom: "Define repositories structured under a common web or filesystem path."
Simplified: "Sync smart proxy content directly from upstream repositories by selecting the desired products."
RHUI: "Define RHUI repository paths with guided steps."

The simplified description feels different, but I can't think of a succinct way to describe it starting with "Define...".

File: webpack/scenes/AlternateContentSources/Create/Steps/SelectSource.js
Comment (sjha4) [inline webpack/scenes/AlternateContentSources/Create/Steps/SelectSource.js:41]: This layout makes it look good.
<Grid hasGutter>
           <GridItem span={4} rowSpan={2}>

File: webpack/scenes/AlternateContentSources/Create/Steps/SelectSource.js
Comment (sjha4) [inline webpack/scenes/AlternateContentSources/Create/Steps/SelectSource.js:65]: Translations.

**Comment by jeremylenz** [comment]
> the alternate content source actually becomes the main content source. It's more for importing content that is closer to you. Like an override.

Wow, TIL!

Maybe it's worth adding a sentence to the top blurb: "Content will be synced from the alternate content source first, then the original source if the ACS is not reachable."

**Comment by ianballou** [comment]
I have one last request, can we completely capitalize RHUI on the index page?

image

**Review by ianballou — APPROVED** [review]
I'm cool with this! Thanks!

File: webpack/scenes/AlternateContentSources/Create/Steps/ACSCredentials.js
Comment (sjha4) [inline webpack/scenes/AlternateContentSources/Create/Steps/ACSCredentials.js:36]: Can you add translations here instead of ```description={__(description)}``` below.

File: webpack/scenes/AlternateContentSources/Create/Steps/AcsUrlPaths.js
Comment (sjha4) [inline webpack/scenes/AlternateContentSources/Create/Steps/AcsUrlPaths.js:46]: Same here for translations. These dynamic translations won't work. We can only translate static strings.

**Review by sjha4 — APPROVED** [review]
Awesome work @lfu . Ack 👍🏼

**Pull request merged**
